### ISSUE TITLE
Integrate with Gatsby

### ISSUE THREAD

**Issue by prateekrastogi** [prevwong/craft.js#9]
**Describe the bug**
It needs to integrate with Gatsby.js to fetch Graphql endpoints on Server Side, with a plugin panel, to make complete websites using it.

**To Reproduce**
1. Go to 'https://www.gatsbyjs.org/'
2. Click on 'Documentation'
3. Scroll down to 'Benefits'
4. "Analyze the scope of craft.js core"

**Expected behavior**
Well, you can not still craft.js a fully-fledged modern react websites without disintegrating craft.js core to inject `modern design system` such as antd, `antd-mobile`, material-ui, & `mapboxgl`  

**Screenshots**
Craft.js might have future but it needs to be extra thin editor leveraging the best of something called network effect in the age of internet. 

**Additional context**
As of now, yours code type system is little too coupled for loosening those bounds.

### Your environment
Web Computing
| Software         | Version(s) |
| ---------------- | ---------- |
| craft.js           |
| React            |
| TypeScript       |
| Browser          |
| npm/Yarn         |
| Operating System | Windows

**Comment by prevwong** [comment]
I'm not too sure if I understand completely of what you're trying to propose. There should not be an issue with using those "modern design systems" - the demos are using `material-ui` , but I assume you're referring to something else.

Perhaps it would help if you could provide an example what you're trying to achieve that is not yet possible with Craft.js.

**Comment by prateekrastogi** [comment]
Since you choose to ignore overall context of my issue including heading and got fixated on `modern design systems`. Lets me clear that issue first with examples of Hallway Usability Test:

* https://craft.js.org/r/docs/guides/basic-tutorial#user-components
* Well, "you can not still" craft.js a  `A React Framework for building extensible drag and drop page editors` without Toolbox importing hard dependencies such as this `import { Box, Typography, Grid, Button as MaterialButton } from "@material-ui/core";`  

Or to summarize, your page editor is tightly coupled to the page editing components it statically imports. In other words, we need runtime loading for such components instead of "static compilation".

**Comment by prevwong** [comment]
Look, I did not ignore your overall context. I literally asked for clarification because I did not understand it.

If you're talking about dynamic imports of components, it is currently out of scope for this project. You could try using existing solutions like the ones provided by next.js.

**Comment by prateekrastogi** [comment]
ok

**Comment by ackvf** [comment]
Hey @prateekrastogi , I am just learning this library and trying to make a PoC editor. Your concern seems quite disturbing and might actually affect my decision to stay with this library if it proves to be impractical for a very specific use case I have. Since it's quite hard to understand, would you mind explaining the obstacles you have discovered in layman's terms?

Also I can't seem to follow your _steps to reproduce_.

**Comment by prateekrastogi** [comment]
It must use canvas for drag and drop editor for large scale use-cases. It seems to be most major flaw with craft.js.

**Comment by prevwong** [comment]
@prateekrastogi That made no sense whatsoever 😂 Good job doing that 4 times in a row!

**Comment by ackvf** [comment]
@prateekrastogi Did you mean Canvas craft.js component or canvas html element? I am really not sure what you mean.
